detox/src/devices/DeviceRegistry.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+const plockfile = require('proper-lockfile');
+const _ = require('lodash');
+const retry = require('../utils/retry');
+const LOCK_FILE = './device.registry.state.lock';
+
+class DeviceRegistry {
+
+  constructor({getDeviceIdsByType, maxTestRunners = 1, createDevice}) {
+    this.getDeviceIdsByType = getDeviceIdsByType;
+    this.maxTestRunners = maxTestRunners;
+    this.createDevice = createDevice;
+    createEmptyLockFileIfNeeded();
+  }
+
+  async getDevice(deviceType) {
+    await retry(() => plockfile.lockSync(LOCK_FILE));
```

Comment:
we may want to add Number.MAX_SAFE_INTEGER as retry amount, and reduce the exponential backoff timeouts to minimum.